Tests/scripts/hook_validations/script.py

```diff
@@ -0,0 +1,101 @@
+import re
+import yaml
+
+from Tests.test_utils import print_error, run_git_command
+
+
+class ScriptValidator(object):
+    """ScriptValidator is designed to validate the correctness of the file structure we enter to content repo. And
+        also try to catch possible Backward compatibility breaks due to the preformed changes.
+
+    Attributes:
+       _is_valid (bool): the attribute which saves the valid/in-valid status of the current file.
+       file_path (str): the path to the file we are examining at the moment.
+       change_string (string): the change string for the examined script.
+       yaml_data (dict): the data of the script in dictionary format.
+    """
+    def __init__(self, file_path, check_git=True):
+        self._is_valid = True
+        self.file_path = file_path
+
+        if check_git:
+            self.change_string = run_git_command("git diff HEAD {0}".format(self.file_path))
+            with open(file_path, 'r') as file_data:
+                self.yaml_data = yaml.safe_load(file_data)
```

Comment:
if `check_git` is False, self.yaml_data will not be defined for all class methods